docusaurus/docs/e2e-test-a-plugin/setup-resources.md

```diff
@@ -0,0 +1,83 @@
+---
+id: setup-resources
+title: Configure the resources you need
+description: Configure the resources you need
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - testing
+  - e2e
+  - data-source
+  - preparation
+sidebar_position: 60
+---
+
+# Introduction
+
+In many cases, you'll need certain resources to be configured in Grafana before you can run your e2e tests. This guide covers how to setup the resources that you need.
+
+## Test isolation
+
+[Test isolation](https://playwright.dev/docs/browser-contexts#what-is-test-isolation) is a central part of Playwright testing. On this subject, we recommend testing plugin features independently and not run through advanced flows where certain steps have dependencies to the previous steps.
+
+### A concrete example
+
+Let's say you want to test template variable interpolation in your Data Source plugin. For any interpolation to take place in the DataSource file, there needs to be a template variable defined. Since the goal is to test variable interpolation, we don't want to create the template variable within the test code. A better approach is to open a provisioned dashboard that has a template variable defined already and use that in our test.
+
+In the following example, we navigate to a provisioned dashboard. The dashboard has a multi-valued template variable `env` with the values `test` and `prod`. We add a new panel and set a SQL query that refers to the `env` variable. We then spy on the query data request, asserting that it was called with the expanded values associated with the template variable.
+
+```ts
+test('should expand multi-valued variable before calling backend', async ({
+  readProvision,
+  request,
+  page,
+  selectors,
+  grafanaVersion,
+}) => {
+  const dashboard = await readProvisionDashboard({ fileName: 'variable.json' });
+  const dashboardPage = new DashboardPage({ request, page, selectors, grafanaVersion }, dashboard);
+  await dashboardPage.goto();
+  const panelEditPage = await dashboardPage.addPanel();
+  const queryDataSpy = panelEditPage.waitForQueryDataRequest((request) =>
+    (request.postData() ?? '').includes(`select * from dataset where env in ('test', 'prod')"`)
+  );
+  await page.getByLabel('Query').fill('select * from dataset where env in (${env:singlequote})');
+  await panelEditPage.refreshPanel();
+  await expect(await queryDataSpy).toBeTruthy();
+});
+```
+
+## Provision any required resources
+
+You may use [provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/) to configure resources such as dashboards and data sources.
+
+> If running e2e tests in CI requires provisioning, you may need to remove the `provisioning` folder from your plugin's `.gitignore` file. Be careful not to commit secrets to public repositories. Use [environment variable](https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables) interpolation for sensitive data.
```

Comment:
These feel like separate points. Displaying them is awkward
```suggestion
:::note
If running e2e tests in CI requires provisioning, you may need to remove the `provisioning` folder from your plugin's `.gitignore` file.
:::
:::danger
Be careful not to commit secrets to public repositories. Use [environment variable](https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables) interpolation for sensitive data.
:::
```

These read like separate points of differing levels of concern. A slightly awkward display, but generally we dont use > quote blocks